### ISSUE TITLE
TypeError: Cannot read property 'filter' of undefined at ComboBoxComponent

### ISSUE THREAD

**Issue by RobertoDavila** [bitExpert/ng2-combobox#33]
Hi.

I found a issue when you load data from a server and you have filter=true, it displays this error:

<combo-box [listData]="brands" (onSelect)="selectValue()" [localFilter]="true" [displayField]="'brand_name'" [valueField]="'brand_id'" [(ngModel)]="brand"></combo-box>

ERROR TypeError: Cannot read property 'filter' of undefined
    at ComboBoxComponent

On the component I found the issue:

    private loadData() {
        if (!this.remote) {
            if (this.localFilter) {
                this.data = this._initialData.filter((item) => {

If we validate this._initialData as well we will avoid the error:

    private loadData() {
        if (!this.remote) {
            if (this.localFilter && this._initialData) {
                this.data = this._initialData.filter((item) => {

Is it possible that you add this to the code?

Regards.

**Comment by shochdoerfer** [comment]
Good catch. Thank you for this reporting this issue. Any chance you could provide a pull request with the fix?

**Comment by RobertoDavila** [comment]
Hi shoch.

Yes I can fix that, please provide rights for this.

Regards.

**Comment by shochdoerfer** [comment]
You can simply fork the repo, create a new branch, apply the fixes, push the branch and then create a pull request out of our pushed branch.
